Fabric Wallets panel doesn't load correctly on empty extension directory

<!--- Provide a general summary of the issue in the Title above -->

## Description
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug. What were you hoping to achieve? -->
When the Local Fabric generates for the first time, the Fabric Wallets panel doesn't load any tree items and throws an error.
The panel will then behave correctly if refreshed.

```
[2/10/2020 10:45:39 AM] [IMPORTANT] Log files can be found by running the `Developer: Open Logs Folder` command from the palette

[2/10/2020 10:45:39 AM] [INFO] Starting IBM Blockchain Platform Extension
[2/10/2020 10:45:40 AM] [INFO] Attempting to require dependency: grpc
[2/10/2020 10:45:41 AM] [INFO] Restoring command registry
[2/10/2020 10:45:41 AM] [INFO] Registering commands
[2/10/2020 10:45:41 AM] [INFO] Migrating local runtime manager
[2/10/2020 10:45:41 AM] [INFO] Initializing local runtime manager
[2/10/2020 10:45:41 AM] [ERROR] Error displaying Fabric Wallets: Must call setFabricWalletGenerator first!
[2/10/2020 10:45:41 AM] [YEOMAN] generate.cmd
[2/10/2020 10:45:41 AM] [YEOMAN] generate.sh
[2/10/2020 10:45:41 AM] [YEOMAN] is_generated.cmd
[2/10/2020 10:45:41 AM] [YEOMAN] is_generated.sh
[2/10/2020 10:45:41 AM] [YEOMAN] is_running.cmd
[2/10/2020 10:45:41 AM] [YEOMAN] is_running.sh
[2/10/2020 10:45:41 AM] [YEOMAN] kill_chaincode.cmd
[2/10/2020 10:45:41 AM] [YEOMAN] kill_chaincode.sh
[2/10/2020 10:45:41 AM] [YEOMAN] playbook.yml
[2/10/2020 10:45:41 AM] [YEOMAN] requirements.yml
[2/10/2020 10:45:41 AM] [YEOMAN] start.cmd
[2/10/2020 10:45:41 AM] [YEOMAN] start.sh
[2/10/2020 10:45:41 AM] [YEOMAN] stop.cmd
[2/10/2020 10:45:41 AM] [YEOMAN] stop.sh
[2/10/2020 10:45:41 AM] [YEOMAN] teardown.cmd
[2/10/2020 10:45:41 AM] [YEOMAN] teardown.sh
[2/10/2020 10:45:41 AM] [INFO] Execute stored commands in the registry
[2/10/2020 10:45:41 AM] [INFO] IBM Blockchain Platform Extension activated
```

This can be recreated by deleting your extension home (`.fabric-vscode`) directory.

This problem only seems to happen once - when reloading, everything seems to work fine.

We should also write some Cucumber tests to ensure that we know the extension will activate fine when we have an empty extension home directory.

## Expected Behavior
<!--- Tell us what should happen -->

## Actual Behavior
<!--- Tell us what happens instead -->

## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->

## Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- Reproduce this bug include code to reproduce, if relevant -->
1.
2.
3.

## Existing issues
<!-- Have you searched for any existing issues or are their any similar issues that you've found? -->
- [ ] [Stack Overflow issues](https://stackoverflow.com/questions/tagged/ibp-vscode-extension)
- [ ] [GitHub Issues](https://github.com/IBM-Blockchain/blockchain-vscode-extension/issues)

<!-- Please include any links to issues here -->

## Log File

<!-- In Visual Studio Code's command palette, run '>Developer: Open Logs Folder' -->
<!-- Attach/link to the latest nested 'Blockchain.log' file -->

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* IBM Blockchain Platform Extension Version:
* Visual Studio Code version:
* Environment name and version (e.g. Node.js v8.12.0, npm v6.4.1):
* Operating System and version:
* Link to your project:

--- a/packages/blockchain-extension/cucumber/steps/hooks.ts
+++ b/packages/blockchain-extension/cucumber/steps/hooks.ts
@@ -32,7 +32,7 @@ import { WalletAndIdentityHelper } from '../helpers/walletAndIdentityHelper';
 import { GatewayHelper } from '../helpers/gatewayHelper';
 import { EnvironmentHelper } from '../helpers/environmentHelper';
 import { SampleHelper } from '../helpers/sampleHelper';
-import { FabricRuntimeUtil } from 'ibm-blockchain-platform-common';
+import { FabricRuntimeUtil, LogType } from 'ibm-blockchain-platform-common';
 
 // tslint:disable:no-unused-expression
 
@@ -48,72 +48,57 @@ let firstTime: boolean = true; // Flag used for making sure we do some setup onc
 
 module.exports = function(): any {
 
-    this.timeout = {timeout: 120000 * 1000}; // Global timeout - 2 minutes
+    this.timeout = { timeout: 120000 * 1000 }; // Global timeout - 2 minutes
 
     this.Before(this.timeout, async () => {
-        try {
-            if (firstTime) {
-                this.mySandBox = sinon.createSandbox();
-                this.userInputUtilHelper = new UserInputUtilHelper(this.mySandBox);
-                this.smartContractHelper = new SmartContractHelper(this.mySandBox, this.userInputUtilHelper);
-                this.generatedTestsHelper = new GeneratedTestsHelper(this.mySandBox, this.userInputUtilHelper, this.smartContractHelper);
-                this.walletAndIdentityHelper = new WalletAndIdentityHelper(this.mySandBox, this.userInputUtilHelper);
-                this.gatewayHelper = new GatewayHelper(this.mySandBox, this.userInputUtilHelper);
-                this.fabricEnvironmentHelper = new EnvironmentHelper(this.mySandbox, this.userInputUtilHelper);
-                this.sampleHelper = new SampleHelper(this.mySandBox, this.userInputUtilHelper, this.smartContractHelper);
+        if (firstTime) {
+            await TestUtil.storeRuntimesConfig();
+            await TestUtil.storeExtensionDirectoryConfig();
 
-                VSCodeBlockchainOutputAdapter.instance().setConsole(true);
+            const extDir: string = path.join(__dirname, '..', '..', '..', 'cucumber', 'tmp');
 
-                await TestUtil.storeBypassPreReqs();
-                await vscode.workspace.getConfiguration().update(SettingConfigurations.EXTENSION_BYPASS_PREREQS, true, vscode.ConfigurationTarget.Global);
+            await vscode.workspace.getConfiguration().update(SettingConfigurations.EXTENSION_DIRECTORY, extDir, vscode.ConfigurationTarget.Global);
 
-                const extDir: string = path.join(__dirname, '..', '..', '..', 'cucumber', 'tmp');
+            this.mySandBox = sinon.createSandbox();
+            this.userInputUtilHelper = new UserInputUtilHelper(this.mySandBox);
+            this.smartContractHelper = new SmartContractHelper(this.mySandBox, this.userInputUtilHelper);
+            this.generatedTestsHelper = new GeneratedTestsHelper(this.mySandBox, this.userInputUtilHelper, this.smartContractHelper);
+            this.walletAndIdentityHelper = new WalletAndIdentityHelper(this.mySandBox, this.userInputUtilHelper);
+            this.gatewayHelper = new GatewayHelper(this.mySandBox, this.userInputUtilHelper);
+            this.fabricEnvironmentHelper = new EnvironmentHelper(this.mySandbox, this.userInputUtilHelper);
+            this.sampleHelper = new SampleHelper(this.mySandBox, this.userInputUtilHelper, this.smartContractHelper);
 
-                this.userInputUtilHelper.showConfirmationWarningMessageStub.reset();
-                firstTime = false;
+            VSCodeBlockchainOutputAdapter.instance().setConsole(true);
 
-                const extDirExists: boolean = await fs.pathExists(extDir);
-                if (!extDirExists) {
-                    await fs.mkdir(extDir);
-                }
+            await TestUtil.storeBypassPreReqs();
+            await vscode.workspace.getConfiguration().update(SettingConfigurations.EXTENSION_BYPASS_PREREQS, true, vscode.ConfigurationTarget.Global);
 
-                const packageDir: string = path.join(extDir, 'packages');
-                const contractDir: string = path.join(extDir, 'contracts');
-                const environmentsDir: string = path.join(extDir, 'environments');
-                const walletsDir: string = path.join(extDir, 'wallets');
-                const profileDir: string = path.join(extDir, 'profiles');
-                for (const dir of [packageDir, contractDir, environmentsDir, walletsDir, profileDir]) {
-                    const exists: boolean = await fs.pathExists(dir);
+            this.userInputUtilHelper.showConfirmationWarningMessageStub.reset();
+            firstTime = false;
 
-                    if (exists) {
-                        await fs.remove(dir);
-                    }
-                }
-
-                const tmpRepo: string = path.join(extDir, 'repositories');
-
-                const tmpExists: boolean = await fs.pathExists(tmpRepo);
-                if (tmpExists) {
-                    await fs.remove(tmpRepo);
-                }
-                await fs.mkdir(tmpRepo);
-
-                await TestUtil.storeRuntimesConfig();
-                await TestUtil.storeExtensionDirectoryConfig();
-
-                await vscode.workspace.getConfiguration().update(SettingConfigurations.EXTENSION_DIRECTORY, extDir, vscode.ConfigurationTarget.Global);
-
-                await ExtensionUtil.activateExtension();
-
-                try {
-                    await vscode.commands.executeCommand(ExtensionCommands.TEARDOWN_FABRIC, undefined, true, FabricRuntimeUtil.LOCAL_FABRIC);
-                } catch (error) {
-                    // If the Fabric is already torn down, do nothing
-                }
+            const extDirExists: boolean = await fs.pathExists(extDir);
+            if (extDirExists) {
+                await fs.remove(extDir);
             }
-        } catch (error) {
-            // tslint:disable-next-line: no-console
-            console.log(error);
+
+            await fs.ensureDir(extDir);
+
+            await ExtensionUtil.activateExtension();
+
+            await vscode.commands.executeCommand('environmentExplorer.focus');
+
+            // sleep to allow the panels to refresh
+            await ExtensionUtil.sleep(3000);
+
+            // check there were no errors on activation
+            this.userInputUtilHelper.logSpy.should.not.have.been.calledWith(LogType.ERROR);
+            this.userInputUtilHelper.logSpy.resetHistory();
+
+            try {
+                await vscode.commands.executeCommand(ExtensionCommands.TEARDOWN_FABRIC, undefined, true, FabricRuntimeUtil.LOCAL_FABRIC);
+            } catch (error) {
+                // If the Fabric is already torn down, do nothing
+            }
         }
     });
 

--- a/packages/blockchain-extension/cucumber/steps/sample.steps.ts
+++ b/packages/blockchain-extension/cucumber/steps/sample.steps.ts
@@ -38,6 +38,7 @@ module.exports = function(): any {
         const exists: boolean = await fs.pathExists(pathToCheck);
         let webview: SampleView;
         if (!exists) {
+            await fs.ensureDir(pathToCheck);
             webview = await this.sampleHelper.cloneSample(repositoryName, namespace);
 
         } else {

--- a/packages/blockchain-extension/extension/util/ExtensionUtil.ts
+++ b/packages/blockchain-extension/extension/util/ExtensionUtil.ts
@@ -550,6 +550,8 @@ export class ExtensionUtil {
         outputAdapter.log(LogType.INFO, undefined, 'Restoring command registry');
         tempCommandRegistry.restoreCommands();
 
+        FabricConnectionFactory.createFabricWallet();
+
         outputAdapter.log(LogType.INFO, undefined, 'Registering commands');
         const context: vscode.ExtensionContext = GlobalState.getExtensionContext();
         await ExtensionUtil.registerCommands(context);
@@ -592,8 +594,6 @@ export class ExtensionUtil {
                 await vscode.workspace.getConfiguration().update(SettingConfigurations.HOME_SHOW_ON_NEXT_ACTIVATION, false, vscode.ConfigurationTarget.Global);
             }
         }
-
-        FabricConnectionFactory.createFabricWallet();
 
         // Check if there is a newer version of the generator available
         // This needs to be done as a seperate call to make sure the dependencies have been installed

--- a/packages/blockchain-extension/test/util/ExtensionUtil.test.ts
+++ b/packages/blockchain-extension/test/util/ExtensionUtil.test.ts
@@ -41,6 +41,7 @@ import { RepositoryRegistry } from '../../extension/registries/RepositoryRegistr
 import { RepositoryRegistryEntry } from '../../extension/registries/RepositoryRegistryEntry';
 import * as openTransactionViewCommand from '../../extension/commands/openTransactionViewCommand';
 import { LocalEnvironment } from '../../extension/fabric/environments/LocalEnvironment';
+import { FabricConnectionFactory } from '../../extension/fabric/FabricConnectionFactory';
 
 const should: Chai.Should = chai.should();
 chai.use(sinonChai);
@@ -1051,6 +1052,7 @@ describe('ExtensionUtil Tests', () => {
         let executeStoredCommandsStub: sinon.SinonStub;
         let logSpy: sinon.SinonSpy;
         let getExtensionLocalFabricSettingStub: sinon.SinonStub;
+        let fabricConnectionFactorySpy: sinon.SinonSpy;
 
         beforeEach(() => {
             hasNativeDependenciesInstalledStub = mySandBox.stub(DependencyManager.instance(), 'hasNativeDependenciesInstalled');
@@ -1065,6 +1067,7 @@ describe('ExtensionUtil Tests', () => {
             executeStoredCommandsStub = mySandBox.stub(TemporaryCommandRegistry.instance(), 'executeStoredCommands');
             logSpy = mySandBox.spy(VSCodeBlockchainOutputAdapter.instance(), 'log');
             getExtensionLocalFabricSettingStub = mySandBox.stub(ExtensionUtil, 'getExtensionLocalFabricSetting');
+            fabricConnectionFactorySpy = mySandBox.spy(FabricConnectionFactory, 'createFabricWallet');
         });
 
         it('should install native dependencies if not installed', async () => {
@@ -1099,6 +1102,7 @@ describe('ExtensionUtil Tests', () => {
             getExtensionContextStub.should.have.been.calledOnce;
             registerCommandsStub.should.have.been.calledOnceWithExactly({ hello: 'world' });
             executeStoredCommandsStub.should.have.been.calledOnce;
+            fabricConnectionFactorySpy.should.have.been.called;
         });
 
         it(`shouldn't install native dependencies if they are already installed`, async () => {
@@ -1133,6 +1137,7 @@ describe('ExtensionUtil Tests', () => {
             getExtensionContextStub.should.have.been.calledOnce;
             registerCommandsStub.should.have.been.calledOnceWithExactly({ hello: 'world' });
             executeStoredCommandsStub.should.have.been.calledOnce;
+            fabricConnectionFactorySpy.should.have.been.called;
         });
 
         it(`should rewrite the package.json file if ther native dependencies are installed but there are no activation events`, async () => {
@@ -1167,6 +1172,7 @@ describe('ExtensionUtil Tests', () => {
             getExtensionContextStub.should.have.been.calledOnce;
             registerCommandsStub.should.have.been.calledOnceWithExactly({ hello: 'world' });
             executeStoredCommandsStub.should.have.been.calledOnce;
+            fabricConnectionFactorySpy.should.have.been.called;
         });
 
         it(`should delete local registry entries if not enabled`, async () => {
@@ -1207,6 +1213,7 @@ describe('ExtensionUtil Tests', () => {
             getExtensionContextStub.should.have.been.calledOnce;
             registerCommandsStub.should.have.been.calledOnceWithExactly({ hello: 'world' });
             executeStoredCommandsStub.should.have.been.calledOnce;
+            fabricConnectionFactorySpy.should.have.been.called;
 
             deleteEnvironmentSpy.should.have.been.calledOnceWithExactly(FabricRuntimeUtil.LOCAL_FABRIC, true);
         });

